File: src/main/java/org/example/expert/domain/common/entity/Log.java
package org.example.expert.domain.common.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "log")
@NoArgsConstructor
public class Log extends Timestamped {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String entityName;

    private String action;

    public Log(String entityName, String action) {
        this.entityName = entityName;
        this.action = action;
    }
}

File: src/main/java/org/example/expert/domain/common/entity/Log.kt
package org.example.expert.domain.common.entity

import jakarta.persistence.*

@Entity
@Table(name = "log")
class Log(var entityName: String,
          var action: String) : Timestamped() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
}
